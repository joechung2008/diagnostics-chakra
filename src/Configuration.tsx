import { Table, Text } from "@chakra-ui/react";
import { useMemo } from "react";

const Configuration: React.FC<ConfigurationProps> = ({ config }) => {
  const items = useMemo(
    () =>
      Object.entries(config).reduce<KeyValuePair<string>[]>(
        (previous, [key, value]) => [...previous, { key, value }],
        []
      ),
    [config]
  );

  return (
    <div>
      <Text id="configuration" as="h3" fontSize="lg" fontWeight="semibold">
        Configuration
      </Text>
      <Table.Root aria-labelledby="configuration">
        <Table.Header>
          <Table.Row>
            <Table.ColumnHeader>Key</Table.ColumnHeader>
            <Table.ColumnHeader>Value</Table.ColumnHeader>
          </Table.Row>
        </Table.Header>
        <Table.Body>
          {items.map((item) => (
            <Table.Row key={item.key}>
              <Table.Cell>{item.key}</Table.Cell>
              <Table.Cell>{item.value}</Table.Cell>
            </Table.Row>
          ))}
        </Table.Body>
      </Table.Root>
    </div>
  );
};

export default Configuration;
