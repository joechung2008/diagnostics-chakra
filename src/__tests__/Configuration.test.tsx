import { describe, it, expect } from "vitest";
import { render } from "@testing-library/react";
import Configuration from "../Configuration";
import { TestWrapper } from "./test-utils";

describe("Configuration", () => {
  it("should render configuration heading", () => {
    const config = { key1: "value1" };

    const { asFragment } = render(
      <TestWrapper>
        <Configuration config={config} />
      </TestWrapper>
    );

    expect(asFragment()).toMatchSnapshot();
  });

  it("should render table with correct headers", () => {
    const config = { key1: "value1" };

    const { asFragment } = render(
      <TestWrapper>
        <Configuration config={config} />
      </TestWrapper>
    );

    expect(asFragment()).toMatchSnapshot();
  });

  it("should render single configuration entry", () => {
    const config = { apiUrl: "https://api.example.com" };

    const { asFragment } = render(
      <TestWrapper>
        <Configuration config={config} />
      </TestWrapper>
    );

    expect(asFragment()).toMatchSnapshot();
  });

  it("should render multiple configuration entries", () => {
    const config = {
      apiUrl: "https://api.example.com",
      timeout: "5000",
      retries: "3",
    };

    const { asFragment } = render(
      <TestWrapper>
        <Configuration config={config} />
      </TestWrapper>
    );

    expect(asFragment()).toMatchSnapshot();
  });

  it("should handle empty config object", () => {
    const config = {};

    const { asFragment } = render(
      <TestWrapper>
        <Configuration config={config} />
      </TestWrapper>
    );

    expect(asFragment()).toMatchSnapshot();
  });

  it("should handle special characters in values", () => {
    const config = { special: "value with spaces & symbols !@#$%" };

    const { asFragment } = render(
      <TestWrapper>
        <Configuration config={config} />
      </TestWrapper>
    );

    expect(asFragment()).toMatchSnapshot();
  });

  it("should render table with correct accessibility attributes", () => {
    const config = { key1: "value1" };

    const { asFragment } = render(
      <TestWrapper>
        <Configuration config={config} />
      </TestWrapper>
    );

    expect(asFragment()).toMatchSnapshot();
  });
});
