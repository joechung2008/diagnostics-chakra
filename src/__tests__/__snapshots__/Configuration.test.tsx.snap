// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Configuration > should handle empty config object 1`] = `
<DocumentFragment>
  <script
    nonce=""
  >
    ((e,i,s,u,m,a,l,h)=&gt;{let d=document.documentElement,w=["light","dark"];function p(n){(Array.isArray(e)?e:[e]).forEach(y=&gt;{let k=y==="class",S=k&&a?m.map(f=&gt;a[f]||f):m;k?(d.classList.remove(...S),d.classList.add(a&&a[n]?a[n]:n)):d.setAttribute(y,n)}),R(n)}function R(n){h&&w.includes(n)&&(d.style.colorScheme=n)}function c(){return window.matchMedia("(prefers-color-scheme: dark)").matches?"dark":"light"}if(u)p(u);else try{let n=localStorage.getItem(i)||s,y=l&&n==="system"?c():n;p(y)}catch(n){}})("class","theme","system",null,["light","dark"],null,true,true)
  </script>
  <div>
    <h3
      class="css-okktu3"
      id="configuration"
    >
      Configuration
    </h3>
    <table
      aria-labelledby="configuration"
      class="chakra-table__root css-a2tetu"
    >
      <thead
        class="chakra-table__header"
      >
        <tr
          class="chakra-table__row css-11ot24d"
        >
          <th
            class="chakra-table__columnHeader css-kim7sg"
          >
            Key
          </th>
          <th
            class="chakra-table__columnHeader css-kim7sg"
          >
            Value
          </th>
        </tr>
      </thead>
      <tbody
        class="chakra-table__body"
      />
    </table>
  </div>
</DocumentFragment>
`;

exports[`Configuration > should handle special characters in values 1`] = `
<DocumentFragment>
  <script
    nonce=""
  >
    ((e,i,s,u,m,a,l,h)=&gt;{let d=document.documentElement,w=["light","dark"];function p(n){(Array.isArray(e)?e:[e]).forEach(y=&gt;{let k=y==="class",S=k&&a?m.map(f=&gt;a[f]||f):m;k?(d.classList.remove(...S),d.classList.add(a&&a[n]?a[n]:n)):d.setAttribute(y,n)}),R(n)}function R(n){h&&w.includes(n)&&(d.style.colorScheme=n)}function c(){return window.matchMedia("(prefers-color-scheme: dark)").matches?"dark":"light"}if(u)p(u);else try{let n=localStorage.getItem(i)||s,y=l&&n==="system"?c():n;p(y)}catch(n){}})("class","theme","system",null,["light","dark"],null,true,true)
  </script>
  <div>
    <h3
      class="css-okktu3"
      id="configuration"
    >
      Configuration
    </h3>
    <table
      aria-labelledby="configuration"
      class="chakra-table__root css-a2tetu"
    >
      <thead
        class="chakra-table__header"
      >
        <tr
          class="chakra-table__row css-11ot24d"
        >
          <th
            class="chakra-table__columnHeader css-kim7sg"
          >
            Key
          </th>
          <th
            class="chakra-table__columnHeader css-kim7sg"
          >
            Value
          </th>
        </tr>
      </thead>
      <tbody
        class="chakra-table__body"
      >
        <tr
          class="chakra-table__row css-11ot24d"
        >
          <td
            class="chakra-table__cell css-1luypx0"
          >
            special
          </td>
          <td
            class="chakra-table__cell css-1luypx0"
          >
            value with spaces & symbols !@#$%
          </td>
        </tr>
      </tbody>
    </table>
  </div>
</DocumentFragment>
`;

exports[`Configuration > should render configuration heading 1`] = `
<DocumentFragment>
  <script
    nonce=""
  >
    ((e,i,s,u,m,a,l,h)=&gt;{let d=document.documentElement,w=["light","dark"];function p(n){(Array.isArray(e)?e:[e]).forEach(y=&gt;{let k=y==="class",S=k&&a?m.map(f=&gt;a[f]||f):m;k?(d.classList.remove(...S),d.classList.add(a&&a[n]?a[n]:n)):d.setAttribute(y,n)}),R(n)}function R(n){h&&w.includes(n)&&(d.style.colorScheme=n)}function c(){return window.matchMedia("(prefers-color-scheme: dark)").matches?"dark":"light"}if(u)p(u);else try{let n=localStorage.getItem(i)||s,y=l&&n==="system"?c():n;p(y)}catch(n){}})("class","theme","system",null,["light","dark"],null,true,true)
  </script>
  <div>
    <h3
      class="css-okktu3"
      id="configuration"
    >
      Configuration
    </h3>
    <table
      aria-labelledby="configuration"
      class="chakra-table__root css-a2tetu"
    >
      <thead
        class="chakra-table__header"
      >
        <tr
          class="chakra-table__row css-11ot24d"
        >
          <th
            class="chakra-table__columnHeader css-kim7sg"
          >
            Key
          </th>
          <th
            class="chakra-table__columnHeader css-kim7sg"
          >
            Value
          </th>
        </tr>
      </thead>
      <tbody
        class="chakra-table__body"
      >
        <tr
          class="chakra-table__row css-11ot24d"
        >
          <td
            class="chakra-table__cell css-1luypx0"
          >
            key1
          </td>
          <td
            class="chakra-table__cell css-1luypx0"
          >
            value1
          </td>
        </tr>
      </tbody>
    </table>
  </div>
</DocumentFragment>
`;

exports[`Configuration > should render multiple configuration entries 1`] = `
<DocumentFragment>
  <script
    nonce=""
  >
    ((e,i,s,u,m,a,l,h)=&gt;{let d=document.documentElement,w=["light","dark"];function p(n){(Array.isArray(e)?e:[e]).forEach(y=&gt;{let k=y==="class",S=k&&a?m.map(f=&gt;a[f]||f):m;k?(d.classList.remove(...S),d.classList.add(a&&a[n]?a[n]:n)):d.setAttribute(y,n)}),R(n)}function R(n){h&&w.includes(n)&&(d.style.colorScheme=n)}function c(){return window.matchMedia("(prefers-color-scheme: dark)").matches?"dark":"light"}if(u)p(u);else try{let n=localStorage.getItem(i)||s,y=l&&n==="system"?c():n;p(y)}catch(n){}})("class","theme","system",null,["light","dark"],null,true,true)
  </script>
  <div>
    <h3
      class="css-okktu3"
      id="configuration"
    >
      Configuration
    </h3>
    <table
      aria-labelledby="configuration"
      class="chakra-table__root css-a2tetu"
    >
      <thead
        class="chakra-table__header"
      >
        <tr
          class="chakra-table__row css-11ot24d"
        >
          <th
            class="chakra-table__columnHeader css-kim7sg"
          >
            Key
          </th>
          <th
            class="chakra-table__columnHeader css-kim7sg"
          >
            Value
          </th>
        </tr>
      </thead>
      <tbody
        class="chakra-table__body"
      >
        <tr
          class="chakra-table__row css-11ot24d"
        >
          <td
            class="chakra-table__cell css-1luypx0"
          >
            apiUrl
          </td>
          <td
            class="chakra-table__cell css-1luypx0"
          >
            https://api.example.com
          </td>
        </tr>
        <tr
          class="chakra-table__row css-11ot24d"
        >
          <td
            class="chakra-table__cell css-1luypx0"
          >
            timeout
          </td>
          <td
            class="chakra-table__cell css-1luypx0"
          >
            5000
          </td>
        </tr>
        <tr
          class="chakra-table__row css-11ot24d"
        >
          <td
            class="chakra-table__cell css-1luypx0"
          >
            retries
          </td>
          <td
            class="chakra-table__cell css-1luypx0"
          >
            3
          </td>
        </tr>
      </tbody>
    </table>
  </div>
</DocumentFragment>
`;

exports[`Configuration > should render single configuration entry 1`] = `
<DocumentFragment>
  <script
    nonce=""
  >
    ((e,i,s,u,m,a,l,h)=&gt;{let d=document.documentElement,w=["light","dark"];function p(n){(Array.isArray(e)?e:[e]).forEach(y=&gt;{let k=y==="class",S=k&&a?m.map(f=&gt;a[f]||f):m;k?(d.classList.remove(...S),d.classList.add(a&&a[n]?a[n]:n)):d.setAttribute(y,n)}),R(n)}function R(n){h&&w.includes(n)&&(d.style.colorScheme=n)}function c(){return window.matchMedia("(prefers-color-scheme: dark)").matches?"dark":"light"}if(u)p(u);else try{let n=localStorage.getItem(i)||s,y=l&&n==="system"?c():n;p(y)}catch(n){}})("class","theme","system",null,["light","dark"],null,true,true)
  </script>
  <div>
    <h3
      class="css-okktu3"
      id="configuration"
    >
      Configuration
    </h3>
    <table
      aria-labelledby="configuration"
      class="chakra-table__root css-a2tetu"
    >
      <thead
        class="chakra-table__header"
      >
        <tr
          class="chakra-table__row css-11ot24d"
        >
          <th
            class="chakra-table__columnHeader css-kim7sg"
          >
            Key
          </th>
          <th
            class="chakra-table__columnHeader css-kim7sg"
          >
            Value
          </th>
        </tr>
      </thead>
      <tbody
        class="chakra-table__body"
      >
        <tr
          class="chakra-table__row css-11ot24d"
        >
          <td
            class="chakra-table__cell css-1luypx0"
          >
            apiUrl
          </td>
          <td
            class="chakra-table__cell css-1luypx0"
          >
            https://api.example.com
          </td>
        </tr>
      </tbody>
    </table>
  </div>
</DocumentFragment>
`;

exports[`Configuration > should render table with correct accessibility attributes 1`] = `
<DocumentFragment>
  <script
    nonce=""
  >
    ((e,i,s,u,m,a,l,h)=&gt;{let d=document.documentElement,w=["light","dark"];function p(n){(Array.isArray(e)?e:[e]).forEach(y=&gt;{let k=y==="class",S=k&&a?m.map(f=&gt;a[f]||f):m;k?(d.classList.remove(...S),d.classList.add(a&&a[n]?a[n]:n)):d.setAttribute(y,n)}),R(n)}function R(n){h&&w.includes(n)&&(d.style.colorScheme=n)}function c(){return window.matchMedia("(prefers-color-scheme: dark)").matches?"dark":"light"}if(u)p(u);else try{let n=localStorage.getItem(i)||s,y=l&&n==="system"?c():n;p(y)}catch(n){}})("class","theme","system",null,["light","dark"],null,true,true)
  </script>
  <div>
    <h3
      class="css-okktu3"
      id="configuration"
    >
      Configuration
    </h3>
    <table
      aria-labelledby="configuration"
      class="chakra-table__root css-a2tetu"
    >
      <thead
        class="chakra-table__header"
      >
        <tr
          class="chakra-table__row css-11ot24d"
        >
          <th
            class="chakra-table__columnHeader css-kim7sg"
          >
            Key
          </th>
          <th
            class="chakra-table__columnHeader css-kim7sg"
          >
            Value
          </th>
        </tr>
      </thead>
      <tbody
        class="chakra-table__body"
      >
        <tr
          class="chakra-table__row css-11ot24d"
        >
          <td
            class="chakra-table__cell css-1luypx0"
          >
            key1
          </td>
          <td
            class="chakra-table__cell css-1luypx0"
          >
            value1
          </td>
        </tr>
      </tbody>
    </table>
  </div>
</DocumentFragment>
`;

exports[`Configuration > should render table with correct headers 1`] = `
<DocumentFragment>
  <script
    nonce=""
  >
    ((e,i,s,u,m,a,l,h)=&gt;{let d=document.documentElement,w=["light","dark"];function p(n){(Array.isArray(e)?e:[e]).forEach(y=&gt;{let k=y==="class",S=k&&a?m.map(f=&gt;a[f]||f):m;k?(d.classList.remove(...S),d.classList.add(a&&a[n]?a[n]:n)):d.setAttribute(y,n)}),R(n)}function R(n){h&&w.includes(n)&&(d.style.colorScheme=n)}function c(){return window.matchMedia("(prefers-color-scheme: dark)").matches?"dark":"light"}if(u)p(u);else try{let n=localStorage.getItem(i)||s,y=l&&n==="system"?c():n;p(y)}catch(n){}})("class","theme","system",null,["light","dark"],null,true,true)
  </script>
  <div>
    <h3
      class="css-okktu3"
      id="configuration"
    >
      Configuration
    </h3>
    <table
      aria-labelledby="configuration"
      class="chakra-table__root css-a2tetu"
    >
      <thead
        class="chakra-table__header"
      >
        <tr
          class="chakra-table__row css-11ot24d"
        >
          <th
            class="chakra-table__columnHeader css-kim7sg"
          >
            Key
          </th>
          <th
            class="chakra-table__columnHeader css-kim7sg"
          >
            Value
          </th>
        </tr>
      </thead>
      <tbody
        class="chakra-table__body"
      >
        <tr
          class="chakra-table__row css-11ot24d"
        >
          <td
            class="chakra-table__cell css-1luypx0"
          >
            key1
          </td>
          <td
            class="chakra-table__cell css-1luypx0"
          >
            value1
          </td>
        </tr>
      </tbody>
    </table>
  </div>
</DocumentFragment>
`;
